ion-hash/src/representation.rs

```diff
@@ -26,7 +26,8 @@ pub(crate) trait RepresentationEncoder {
             IonType::Symbol => self.write_repr_symbol(elem.as_symbol())?,
             IonType::String => self.write_repr_string(elem.as_string())?,
             IonType::Clob | IonType::Blob => self.write_repr_blob(elem.as_lob())?,
-            IonType::List | IonType::SExpression => self.write_repr_seq(elem.as_sequence())?,
+            IonType::List => self.write_repr_list(elem.as_list())?,
+            IonType::SExpression => self.write_repr_sexp(elem.as_sexp())?,
```

Comment:
🗺️ The changes in `ion-hash` were made to accommodate splitting the monolithic `Sequence` type into `List` and `SExp` types.

Reply:
Should we consider similar ergonomics for `Blob` and `Clob`?

```suggestion
            IonType::Clob | IonType::Blob => self.write_repr_blob(elem.as_lob())?,
```

Reply:
> Should we consider similar ergonomics for `Blob` and `Clob`?
> 
> ```
>             IonType::Clob | IonType::Blob => self.write_repr_blob(elem.as_lob())?,
> ```

`Element`  offers `as_lob`, `as_clob`, and `as_blob` methods:
https://github.com/amazon-ion/ion-rust/blob/5c88f8eb712aeee185bc03cbbd4be1171bed4b0a/src/value/owned.rs#L677-L696

Similarly, it also has `as_sequence`, `as_list`, and `as_sexp` methods:
https://github.com/amazon-ion/ion-rust/blob/5c88f8eb712aeee185bc03cbbd4be1171bed4b0a/src/value/owned.rs#L698-L718